### PR TITLE
Add support for local blob storage

### LINKED ISSUE #1023: Add support for local blob storage
Currently JabbR only supports blob storage via Windows Azure.

Adding support for writing attachments to local (or UNC) paths would allow for deployments that do not leverage Windows Azure Blob storage but rather more traditional means of serving static content (CDNs with Origin Server).
- I have updated the work done for #1015 to support the Local storage option in the status page jabbr - status
- Updated the Administration page to reflect the new configuration values (LocalBlobStoragePath and LocalBlobStorageUriPrefix) jabbr - administration page
- Added several string resources in LanguageResources.resx to support the  Administration page additions 

Here is a screenshot of it in action: 
jabbr

### PR DESCRIPTION
**Pull request by ElanHasson**
This commit resolves #1023 

Added support for writing attachments to local (or UNC) paths.
- I have updated the work done for #1015 to support the Local storage
  option in the status page jabbr -
  status
- Updated the Administration page to reflect the new configuration
  values:
  -- LocalBlobStoragePath: This is the local path where files will be
  written to  (e.g: d:\files).
  -- LocalBlobStorageUriPrefix: This is the URI Prefix for the files to be
  served from (e.g: http://static.jabbr.net/files/)

Both values must be configured for IsValid() to return true on the
LocalBlobStorageHandler. Only one of the providers
(LocalBlobStorageHandler or AzureBlobStorageHandler) should be
configured at one time.

A separate site must be configured in IIS or another web server in order
to serve the static file uploads, offloading it from the JabbR server
and allowing for a much larger CDN story to follow without additional
enhancements to JabbR.

When directories get too large, administrators may change the
LocalBlobStoragePath and LocalBlobStorageUriPrefix configuration values
to change where files are written from without impacting already
uploaded files.

jabbr - administration 1
- Added several string resources in LanguageResources.resx to support
  the  Administration page additions

Here is a screenshot of it in action:

jabbr

### CODE REVIEW

File: JabbR/LanguageResources.resx [hunk 1]
```diff
@@ -340,9 +358,9 @@
     <value>FAQ</value>
   </data>
   <data name="Client_FAQMessage" xml:space="preserve">
-    <value>&lt;p&gt;Click on a user to send message.&lt;/p&gt;
```
Comment (ElanHasson) [inline JabbR/LanguageResources.resx:343]: These were changed automatically by the resource designer tool.

File: JabbR/LanguageResources.resx [hunk 2]
```diff
@@ -340,9 +358,9 @@
...
-&lt;p&gt;Type #roomname to create a link to a room.&lt;/p&gt;
```
Comment (ElanHasson) [inline JabbR/LanguageResources.resx:344]: These were changed automatically by the resource designer tool.

File: JabbR/LanguageResources.resx [hunk 3]
```diff
@@ -1094,7 +1112,7 @@ Use ? or type /? to display the FAQ and list of commands.</value>
     <value>The link used has expired.</value>
   </data>
   <data name="Account_ResetFailureMessage" xml:space="preserve">
-    <value>&lt;p&gt;Your password reset was unsuccessful: {0}.&lt;/p&gt;
```
Comment (ElanHasson) [inline JabbR/LanguageResources.resx:1097]: These were changed automatically by the resource designer tool.

File: JabbR/LanguageResources.resx [hunk 4]
```diff
@@ -1107,14 +1125,14 @@ Use ? or type /? to display the FAQ and list of commands.</value>
     <value>&lt;a href="{0}"&gt;Reset your password&lt;/a&gt; if you've forgotten or lost your login information.</value>
   </data>
   <data name="Account_ResetPasswordSuccessMessage" xml:space="preserve">
-    <value>&lt;p&gt;A password reset email has been sent to the email address registered for user &lt;strong&gt;{0}&lt;/strong&gt;.&lt;/p&gt;
```
Comment (ElanHasson) [inline JabbR/LanguageResources.resx:1110]: These were changed automatically by the resource designer tool.

File: JabbR/LanguageResources.resx [hunk 5]
```diff
@@ -1107,14 +1125,14 @@ Use ? or type /? to display the FAQ and list of commands.</value>
...
   <data name="Account_ResetPasswordSuccessMessage" xml:space="preserve">
-    <value>&lt;p&gt;A password reset email has been sent to the email address registered for user &lt;strong&gt;{0}&lt;/strong&gt;.&lt;/p&gt;
+    <value>&lt;p&gt;A password reset email has been sent to the email address registered for user &lt;strong&gt;{0}&lt;/strong&gt;.&lt;/p&gt;
 &lt;p&gt;If you don't see this email in your inbox within 15 minutes, look for it in your junk-mail folder.&lt;/p&gt;</value>
   </data>
   <data name="Account_ResetRequest" xml:space="preserve">
     <value>Request reset</value>
   </data>
   <data name="Account_ResetSuccessMessage" xml:space="preserve">
-    <value>&lt;p&gt;Your password has been reset successfully. &lt;/p&gt;
```
Comment (ElanHasson) [inline JabbR/LanguageResources.resx:1117]: These were changed automatically by the resource designer tool.

File: JabbR/LanguageResources.resx [hunk 6]
```diff
@@ -773,7 +791,7 @@
     <value>(plus {0} hidden... click to expand)</value>
   </data>
   <data name="Chat_InitialMessages" xml:space="preserve">
-    <value>Welcome to JabbR
```
Comment (ElanHasson) [inline JabbR/LanguageResources.resx:776]: These were changed automatically by the resource designer tool.

File: JabbR/LanguageResources.resx [hunk 7]
```diff
@@ -340,9 +358,9 @@
     <value>FAQ</value>
   </data>
   <data name="Client_FAQMessage" xml:space="preserve">
-    <value>&lt;p&gt;Click on a user to send message.&lt;/p&gt;
-&lt;p&gt;Type #roomname to create a link to a room.&lt;/p&gt;
-&lt;p&gt;Use #test for testing.&lt;/p&gt;
```
Comment (ElanHasson) [inline JabbR/LanguageResources.resx:345]: These were changed automatically by the resource designer tool.

**Comment by davidfowl** [comment]
Change the name LocalBlobXX to LocalFileSystem

File: JabbR/UploadHandlers/LocalBlobStorageHandler.cs [hunk 8]
```diff
@@ -0,0 +1,62 @@
...
+        }
+
+        public LocalBlobStorageHandler(ApplicationSettings settings)
+        {
+            _settingsFunc = () => settings;
+        }
+
+        public bool IsValid(string fileName, string contentType)
+        {
+            // Blob storage can handle any content
```
Comment (davidfowl) [inline JabbR/UploadHandlers/LocalBlobStorageHandler.cs]: Evaluate the setttings func once

``` C#
var settings = _settingsFunc();

// Use settings here
```

File: JabbR/UploadHandlers/LocalBlobStorageHandler.cs [hunk 9]
```diff
@@ -0,0 +1,62 @@
...
+            return (!String.IsNullOrEmpty(_settingsFunc().LocalBlobStoragePath) &&
+                    !String.IsNullOrEmpty(_settingsFunc().LocalBlobStorageUriPrefix));
+        }
+
+        public async Task<UploadResult> UploadFile(string fileName, string contentType, Stream stream)
+        {
+            // Randomize the filename everytime so we don't overwrite files
```
Comment (davidfowl) [inline JabbR/UploadHandlers/LocalBlobStorageHandler.cs]: Evaluate the settings func once.

File: JabbR/UploadHandlers/LocalFileSystemStorageHandler.cs [hunk 10]
```diff
@@ -0,0 +1,66 @@
...
+        public async Task<UploadResult> UploadFile(string fileName, string contentType, Stream stream)
+        {
+            var settings = _settingsFunc();
+
+            // Randomize the filename everytime so we don't overwrite files
+            string randomFile = Path.GetFileNameWithoutExtension(fileName) +
+                                "_" +
+                                Guid.NewGuid().ToString().Substring(0, 4) + Path.GetExtension(fileName);
+
+            if (!Directory.Exists(_settingsFunc().LocalFileSystemStoragePath))
```
Comment (davidfowl) [inline JabbR/UploadHandlers/LocalFileSystemStorageHandler.cs]: You forgot to actually use the settings variable.

**Comment by ElanHasson** [comment]
Good to go.. let me know if you would like anything else corrected.

**Pull request merged**